### PR TITLE
add mean function, get current data function.

### PR DESCRIPTION
**Pull request by ecomerford**
@georgf in this PR, I have:

- added a `getMean` function to calculate the mean.
- added a display of the mean in the distribution view.
- added a `getCurrentData` function that retrieves data dynamically.

In the future (i.e. hopefully later this week), these functions will all update based on changes to the state (`activeMetric`, `activeChannel`, and `activeVersion`, specifically); however, as discussed, I am still working through the logistics of lifecycle methods. 🙂 

l am also working on a version of the math functions that receives a data property, so that we can specify if we'd like to use the current or previous version data; this will set us up to be able to calculate statistics from previous versions, as well as the % change of those statistics.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,28 +17,65 @@ class App extends React.Component {
...
+        var metrics = probeData.map(item => item.name);
+        this.setState({
+          probeInfo: probeData,
+          metricOptions: metrics,
+        });
+      });
+  }
+
+  getCurrentData = () => {
+    fetch("https://mozilla.github.io/mdv2/data/" + this.state.activeMetric + "_" + this.state.activeChannel + "_" + this.state.activeVersion + ".json")
```
Comment (georgf) [inline src/App.js]: Small thing: This line is rather long and goes outside of the review view width, can you break this up a bit at say max. 100 characters?
```
fetch("..." +
         "...")
  .then(...)
```

File: src/App.js [hunk 2]
```diff
@@ -17,28 +17,65 @@ class App extends React.Component {
...
+    let values = this.state.currentData.map(item => item.count);
+    buckets = buckets.concat([this.getLastBucketUpper()]);
+    let totalHits = 0,
+        bucketHits = 0;
+    let linearTerm = (buckets[buckets.length - 1] - buckets[buckets.length -2]) / 2;
+    let exponentialFactor = Math.sqrt(buckets[buckets.length - 1] / buckets[buckets.length - 2]);
+    let useLinearBuckets = this.kind === "linear" || this.kind === "flag" || this.kind === "boolean" || this.kind === "enumerated";
+    for (let i = 0; i < values.length; i++) {
+      totalHits += values[i];
+      let centralX = useLinearBuckets ? buckets[i] + linearTerm : buckets[i] * exponentialFactor; // find the center of the current bucket
```
Comment (georgf) [inline src/App.js]: Small thing: The comment would be more readable if it's on the preceding line.

File: src/App.js [hunk 3]
```diff
@@ -17,28 +17,65 @@ class App extends React.Component {
...
+  getCurrentData = () => {
+    fetch("https://mozilla.github.io/mdv2/data/" + this.state.activeMetric + "_" + this.state.activeChannel + "_" + this.state.activeVersion + ".json")
+      .then(response => response.json())
+      .then(data => this.setState({currentData: data}));
+  }
+
+  getMean = () => {
+    let buckets = this.state.currentData.map(item => item.start);
+    let values = this.state.currentData.map(item => item.count);
+    buckets = buckets.concat([this.getLastBucketUpper()]);
```
Comment (georgf) [inline src/App.js]: Small thing: Why not have this be part of the `let buckets ...` line?
```
let buckets = ....map(...)
                .concat(...);
```

**Review by georgf — APPROVED** [review]
This makes sense, especially keeping in mind the planned refactoring.

I have some small comments below, feel free to just change those and then merge the PR.

**Comment by ecomerford** [comment]
Thanks for the edits! Should be all set now; I'll merge.

**Pull request merged**